### ISSUE TITLE
Bug: derive_vars_joined - when BY_VARS holds an expression get an ERROR

### ISSUE THREAD

**Issue by millerg23** [pharmaverse/admiral#2289]
### What happened?

Roche user reported they were getting an ERROR when using `derive_vars_joined funtion()` when BY_VARS had an expression.
When expression removed and just select variables it works fine.

### Session Information

```
library(admiral)
library(dplyr)
library(tibble)

ex <- tribble(
  ~USUBJID, ~EXLINKID,
  "1",      "001",
  "1",      "002",
  "2",      "001",
)

fa <- tribble(
  ~USUBJID, ~FALINKID, ~FA,
  "1",      "001",     1,
  "1",      "002",     0,
  "2",      "001",     1,
)

test <- derive_vars_joined(
  ex,
  dataset_add = fa,
  by_vars = exprs(USUBJID, EXLINKID = FALINKID),
  join_type = "all",
  filter_add = FA == 1
)
```
```
Error in `assert_data_frame()` at admiral/R/derive_merged.R:322:2:
! Required variable `FALINKID` is missing
Run `rlang::last_trace()` to see where the error occurred.
```

**Comment by millerg23** [comment]
@bms63 think this needs fixed in this release (0.1.0)

**Comment by bms63** [comment]
We actually test for this scenario? 
![image](https://github.com/pharmaverse/admiral/assets/10111024/1df5a5ab-0283-431f-abcc-9b4282a8a825)

**Comment by bms63** [comment]
The use of `by_vars_left` in the code might be the culprit https://github.com/pharmaverse/admiral/blob/main/R/derive_joined.R

**Comment by millerg23** [comment]
Ah, thanks @bms63.
I had a play around, when I add in `new_vars = FA` it works!!

**Comment by bms63** [comment]
> Ah, thanks @bms63. I had a play around, when I add in `new_vars = FA` it works!!

ummm...but that is not very helpful for user!!

**Comment by millerg23** [comment]
yeah I agree, I didn't mean issue had gone away, but perhaps have narrowed down where issue is!

**Comment by bms63** [comment]
The error message seems to indicate line 521 where `derive_var_merged()` is being used.

![image](https://github.com/pharmaverse/admiral/assets/10111024/1f223381-0fbe-4bc7-a20f-237a2943daca)

**Comment by bms63** [comment]
oh i see @bundfussr is assigned for this issue.  You got the bandwidth Stefan??  Maybe we could get @zdz2101 on this one?

**Comment by bms63** [comment]
![image](https://github.com/pharmaverse/admiral/assets/10111024/cc51d4ad-0656-4ebc-80ab-2af4b1c9bcf1)
